### PR TITLE
Speculative Ideas about Titanic Example

### PR DESCRIPTION
**Pull request by shiffman**
This goes along with https://github.com/ml5js/ml5-library/pull/569. I am rewriting this example with some ideas of how it might be, but as it stands, the code in ml5-library does not (yet) support these ideas. This is mostly for tracking / discussion.

@joeyklee apologies that my fancy autoformat VSCode settings is changing all the whitespace and making track changes hard to follow.

### CODE REVIEW

**Comment by shiffman** [comment]
More to do still but @joeyklee if you want to review, comment, and/or merge I likely won't look at this again until later tomorrow.

**Comment by joeyklee** [comment]
@shiffman - I will merge this into the `generic-neural-network` branch now so we can sync up. Thanks!
